jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigurationMigrator.java

```diff
@@ -164,6 +138,84 @@ private void migrate(String wikiId)
+                authors.setCreator(SuperAdminUserReference.INSTANCE);
+                authors.setContentAuthor(SuperAdminUserReference.INSTANCE);
+                authors.setEffectiveMetadataAuthor(SuperAdminUserReference.INSTANCE);
+                authors.setOriginalMetadataAuthor(SuperAdminUserReference.INSTANCE);
+
+                xwiki.saveDocument(doc, "Set superadmin as creator and author", true, context);
+            }
+        }
+    }
+
+    private void migrate(String wikiId, WikiReference wikiReference)
+        throws XWikiException
+    {
+        XWikiContext context = contextProvider.get();
+        XWiki xwiki = context.getWiki();
+        DocumentReference basicAuthConfigRef =
+            new DocumentReference(BasicAuthJIRAAuthenticatorFactory.BASIC_AUTH_CONFIG_REFERENCE, wikiReference);
+        XWikiDocument basicAuthDoc = xwiki.getDocument(basicAuthConfigRef, context).clone();
+        XWikiDocument jiraServerDoc =
+            xwiki.getDocument(new DocumentReference(wikiId, JIRA, "JIRAConfig"), context).clone();
+        List<BaseObject> jiraServerObjs =
+            jiraServerDoc.getXObjects(new DocumentReference(wikiId, JIRA, "JIRAConfigClass"));
+
+        basicAuthDoc.setHidden(true);
+        DocumentAuthors authors = basicAuthDoc.getAuthors();
+        if (basicAuthDoc.isNew()) {
+            authors.setCreator(SuperAdminUserReference.INSTANCE);
+            authors.setContentAuthor(SuperAdminUserReference.INSTANCE);
+            authors.setEffectiveMetadataAuthor(SuperAdminUserReference.INSTANCE);
+            authors.setOriginalMetadataAuthor(SuperAdminUserReference.INSTANCE);
+        }
+
+        for (BaseObject obj : jiraServerObjs) {
+            if (obj == null) {
+                continue;
+            }
+            String serverId = obj.getStringValue(PROPERTY_ID);
+            String username = obj.getStringValue(PROPERTY_USERNAME);
+            String password = obj.getStringValue(PROPERTY_PASSWORD);
+
+            obj.removeField(PROPERTY_USERNAME);
+            obj.removeField(PROPERTY_PASSWORD);
+
+            if (StringUtils.isEmpty(username)) {
+                // server without authentication
+                obj.setStringValue(PROPERTY_AUTHENTICATION_TYPE, "noAuth");
+            } else {
+                obj.setStringValue(PROPERTY_AUTHENTICATION_TYPE,
+                    BasicAuthJIRAAuthenticatorFactory.HINT);
+                BaseObject basicAuthObj = basicAuthDoc
+                    .newXObject(BasicAuthJIRAAuthenticatorFactory.BASIC_AUTH_DATA_CLASS_REFERENCE, context);
+                basicAuthObj.setStringValue(PROPERTY_ID, serverId);
+                basicAuthObj.setStringValue(PROPERTY_USERNAME, username);
+                basicAuthObj.setStringValue(PROPERTY_PASSWORD, password);
+            }
+        }
+        xwiki.saveDocument(basicAuthDoc, DOC_SAVE_MESSAGE, context);
+        xwiki.saveDocument(jiraServerDoc, DOC_SAVE_MESSAGE, context);
```

Comment:
>  private static final String DOC_SAVE_MESSAGE = "Migrate configration";

There's a typo and it should be more explicit as it's not easy to understand what migration is being referred to. Also it might be better to use the past tense since it's a message about what was done.